File: buildSrc/src/main/java/wolkenschloss/HostExtension.java
package wolkenschloss;

import org.gradle.api.provider.Property;

public abstract class HostExtension {

    public static final int DEFAULT_CALLBACK_PORT = 9191;

    public void initialize() {
        getHostAddress().convention(IpUtil.getHostAddress());
        getCallbackPort().set(DEFAULT_CALLBACK_PORT);
    }

    public abstract Property<String> getHostAddress();

    public abstract Property<Integer> getCallbackPort();
}

File: buildSrc/src/main/kotlin/wolkenschloss/gradle/testbed/HostExtension.kt
package wolkenschloss.gradle.testbed

import org.gradle.api.provider.Property

abstract class HostExtension {
    fun initialize() {
        hostAddress.convention(IpUtil.hostAddress)
        callbackPort.set(DEFAULT_CALLBACK_PORT)
    }

    abstract val hostAddress: Property<String>
    abstract val callbackPort: Property<Int>

    companion object {
        const val DEFAULT_CALLBACK_PORT = 9191
    }
}